aten/src/ATen/native/cuda/Lerp.cu

```diff
@@ -33,12 +75,56 @@ void lerp_tensor_kernel(at::TensorIteratorBase& iter) {
                           (opmath_t{1} - weight_val_f);
             });
       });
+  }
 }
 
+const char lerp_scalar_name[] = "lerp_scalar_kernel";
 void lerp_scalar_kernel(at::TensorIteratorBase& iter, const c10::Scalar& weight) {
-  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2(
+  auto dtype = iter.common_dtype();
+  if (at::isComplexType(dtype)) {
+#if false // AT_USE_JITERATOR()
+  static const auto lerp_scalar_string = jiterator_stringify(
+      template <typename T>
+      T lerp_scalar_kernel(T self_val, T end_val, int weight) {
+        auto weight_val = weight.to<T>();
+        return (std::abs(weight_val) < 0.5)
+            ? self_val + weight_val * (end_val - self_val)
+            : end_val -
+                (end_val - self_val) * (static_cast<T>(1) - weight_val);
+      }
+  ); // lerp_scalar_string
+  AT_DISPATCH_COMPLEX_TYPES_AND(kComplexHalf, dtype, "lerp_cuda", [&] {
+      jitted_gpu_kernel<
+        /*name=*/ lerp_scalar_name,
+        /*return_dtype=*/ scalar_t,
+        /*common_dtype=*/ scalar_t,
+        /*arity=*/ 2>(
+        iter,
+        lerp_scalar_string,
+        /*scalar_pos=*/ at::cuda::jit::BinaryFuncVariant::NoScalar,
+        /*scalar_val=*/ 0,
+        /*extra_args=*/ std::make_tuple(weight));
```

Comment:
You will have to get a C++ type out of weight (c10::Scalar) before passing it to jiterator function.
Eg. `weight.to<scalar_t>();`